src/lib.rs

```diff
@@ -334,20 +335,16 @@ pub trait Rand : Sized {
 /// A random number generator.
 pub trait Rng {
     /// Return the next random u32.
-    ///
-    /// This rarely needs to be called directly, prefer `r.gen()` to
-    /// `r.next_u32()`.
-    // FIXME #rust-lang/rfcs#628: Should be implemented in terms of next_u64
     fn next_u32(&mut self) -> u32;
 
     /// Return the next random u64.
-    ///
-    /// By default this is implemented in terms of `next_u32`. An
-    /// implementation of this trait must provide at least one of
-    /// these two methods. Similarly to `next_u32`, this rarely needs
-    /// to be called directly, prefer `r.gen()` to `r.next_u64()`.
+    /// 
+    /// This function has a default implementation  of `next_u32`. The
+    /// default implementation should not be used in wrapper types since the
+    /// wrapped RNG may have its own implementation which may be more efficient
+    /// or even produce different results.
     fn next_u64(&mut self) -> u64 {
-        ((self.next_u32() as u64) << 32) | (self.next_u32() as u64)
+        impls::next_u64_via_u32(self)
```

Comment:
This is also a breaking change (was BE, or possibly undefined order)

Reply:
No, it turns out the old code also produced LE values (for me at least).

Reply:
Hmm, having that as a breaking change does not make me happy.

I double-checked your results, and if I add the test on the master branch it fails...

Reply:
Thanks for picking that up. I forgot `IsaacRng` already has it's own impl of `next_u64`! Now I get BE results from the old code too. Ok, will revert.